lib/model.js

```diff
@@ -353,7 +354,10 @@ Model.prototype.$__handleSave = function(options, callback) {
       }
       this.constructor.exists(where, optionsWithCustomValues)
         .then(documentExists => {
-          const matchedCount = !documentExists ? 0 : 1;
+          const matchedCount = documentExists ? 1 : 0;
+          if (matchedCount == 0 && optimisticConcurrency) {
```

Comment:
matchedCount of 0 is equivalent to 

```suggestion
          if (!matchedCount && optimisticConcurrency) {
```

Reply:
We could actually rename it from matchedCount to something like "hasMatchedDocument" or so.